src/pages/Search/SearchPage.tsx

```diff
@@ -215,6 +218,61 @@ function SearchPage({route}: SearchPageProps) {
                             // Custom IS templates are not policy specific, so we don't need to pass a policyID
                             beginExportWithTemplate(template.name, CONST.EXPORT_TEMPLATE_TYPES.INTEGRATIONS, undefined);
                         },
+                        shouldCloseModalOnSelect: true,
+                        shouldCallAfterModalHide: true,
+                    });
+                }
+            }
+
+            // Collate a list of policyIDs from the selected transactions
+            const selectedPolicyIDs = [
+                ...new Set(
+                    Object.values(selectedTransactions)
+                        .map((transaction) => transaction.policyID)
+                        .filter(Boolean),
+                ),
+            ];
+
+            // If all of the transactions are on the same policy, add the policy-level in-app export templates as export options
+            if (selectedPolicyIDs.length === 1) {
```

Comment:
For this, if you have more than 500 expenses and use "Select all matching items", it's possible some of the other non-loaded transactions will be on other policies, even though all the loaded ones are on the same policy.
I don't think it's necessarily a bug though, just putting it out there. I don't think there's an easy solution other than loading all expenses or making a API call that would return the unique policy IDs for a search query, but that's unoptimal.

Reply:
Oh, yeah, good point... We could probably filter them out in the auth command since the report data we load before processing the formulas should have the policyID

Reply:
Filter them out in what Auth command?
In the end, I don't think it's an issue if you export an expense or report with a layout that's defined in a different policy, right? Like, nothing can break, and the worst possible case is that your tags / employee custom fields / GL codes are defined differently, and the output is slightly inconsistent.

Reply:
in `ExportSearchWithTemplate`  but yes agreed it's not an urgent issue - I'll test it tomorrow to see what happens